src/zenml/integrations/tensorflow/materializers/tf_dataset_materializer.py

```diff
@@ -40,8 +43,15 @@ def handle_input(self, data_type: Type[Any]) -> Any:
             A tf.data.Dataset object.
         """
         super().handle_input(data_type)
-        path = os.path.join(self.artifact.uri, DEFAULT_FILENAME)
-        return tf.data.experimental.load(path)
+        temp_dir = tempfile.TemporaryDirectory()
+        try:
+            io_utils.copy_dir(self.artifact.uri, temp_dir.name)
+            path = os.path.join(temp_dir.name, DEFAULT_FILENAME)
+            dataset = tf.data.experimental.load(path)
+        finally:
+            fileio.rmtree(temp_dir.name)
```

Comment:
I think we need to do this flow for basically every materializer right? If so, it might make sense to refactor this into the base materializer

Reply:
Completely agree, I created a ticket for it.